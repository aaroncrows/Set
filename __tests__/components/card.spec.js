import React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import Card from 'src/components/card/card'

let testCard
beforeEach(() => {
  testCard = {
    shape: 'oval',
    color: 'first',
    pattern: 'solid',
    count: 1
  }
})

test('should render correctly', () => {
  const testRender = create(<Card card={testCard} />)

  expect(testRender).toMatchSnapshot()
})

test('should pass the card to onClick', () => {
  const mockClick = jest.fn()
  const testRender = shallow(<Card card={testCard} onCardClick={mockClick}/>)
  testRender.simulate('click')
  const clickArguments = mockClick.mock.calls[0][0]

  expect(clickArguments).toEqual(testCard)
})
