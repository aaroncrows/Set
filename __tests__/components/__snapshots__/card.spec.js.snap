// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<li
  className="card"
  onClick={[Function]}
>
  <span
    className="icons-one color-first icon-leaf"
  />
  <span
    className="pattern"
  >
    solid
  </span>
</li>
`;
